docs/images/architecture.mermaid

```diff
@@ -1,26 +1,75 @@
 sequenceDiagram
 
-MainVerticle->>File System: Set up file system watcher
-
-loop
- File System->>WatcherVerticle: Sends new/updated CSV event
-
- loop
-  WatcherVerticle->>ConverterVerticle: Sends item / audio file conversion request
-  ConverterVerticle->>File System: Writes temporary MP4 audio file
-  ConverterVerticle->>PairtreeVerticle: Sends item with audio file for insertion into Pairtree
-  File System->>PairtreeVerticle: Reads temporary MP4 audio file
-  PairtreeVerticle->>ConverterVerticle: Notifies of successful Pairtree insertion
-  ConverterVerticle->>File System: Deletes temporary MP4 audio file
-  ConverterVerticle->>WatcherVerticle: Sends end of loop notification
- end
-
- loop
-  WatcherVerticle->>PairtreeVerticle: Sends item with video file for insertion into Pairtree
-  File System->>PairtreeVerticle: Reads MP4 video file
-  PairtreeVerticle->>WatcherVerticle: Sends end of loop notification
- end
-
- WatcherVerticle->>File System: Writes modified CSV file
-end
+    participant MainVerticle;
+    participant WatcherVerticle;
+    participant ConverterVerticle;
+    participant FFmpeg;
+    participant PairtreeVerticle;
+    participant File System;
+    participant WaveformVerticle;
+    participant audiowaveform;
+    participant Amazon S3;
 
+    activate MainVerticle
+
+    MainVerticle->>File System: Set up file system watcher
+
+    loop
+        File System->>WatcherVerticle: Sends new/updated CSV event
+        activate WatcherVerticle
+
+        loop for each item
+            alt is audio
+                par Converts to MP4 and inserts into Pairtree
+                    WatcherVerticle->>ConverterVerticle: Sends audio file for conversion to MP3
+                    activate ConverterVerticle
+
+                    ConverterVerticle->>FFmpeg: Invokes FFmpeg
+                    activate FFmpeg
+
+                    FFmpeg->>File System: Writes temporary MP4 audio file
+                    FFmpeg-->>ConverterVerticle: FFmpeg exits
+                    deactivate FFmpeg
+
+                    ConverterVerticle->>PairtreeVerticle: Sends converted audio file for insertion into Pairtree
+                    activate PairtreeVerticle
+
+                    PairtreeVerticle->>File System: Reads temporary MP4 audio file
+                    PairtreeVerticle-->>ConverterVerticle: Notifies of successful Pairtree insertion
+                    deactivate PairtreeVerticle
+
+                    ConverterVerticle->>File System: Deletes temporary MP4 audio file
+                    ConverterVerticle-->>WatcherVerticle: Notifies of successful conversion and insertion
+                    deactivate ConverterVerticle
+                and Generates waveform data and stores on Amazon S3
```

Comment:
Is this supposed to be left indented farther than the others?

Reply:
Based on [the syntax description of the `par ... and` construct](https://mermaid-js.github.io/mermaid/#/sequenceDiagram?id=parallel), it looks right to me (there isn't a concrete example like there is for `alt ... else` or `loop`, but the examples for those constructs use the same indentation pattern that I've used here).

Reply:
Ah, okay... part of the problem, I think, is that alt and par aren't obvious (I don't think one should have to go read about the diagramming spec to know what the components of the diagram are).